app/models/match.rb

```diff
@@ -4,4 +4,28 @@ class Match < ApplicationRecord
 
     validates :home_team, presence: true
     validates :away_team, presence: true
+
+    has_many :player_stats
+    has_many :players, through: :player_stats
+    has_many :stats, through: :player_stats
+
```

Comment:
i'll admit i'm not sure if this is the correct location to do the things below this line

Reply:
a general strategy is to keep business logic out of the model. this helps keep the model lightweight and focused on moving data in / out of the database. the scoring methods you have here would probably be better in service classes.